micro_dl/torch_unet/networks/Unet2D.py

```diff
@@ -207,8 +212,39 @@ def forward(self, x, validate_input=False):
         # output channel collapsing layer
         x = self.terminal_block(x)
 
+        #crop padded inputs to valid regions
+        if pad_nonmultiple_input:
+            x = x[...,0:-pad_shape[1], 0:-pad_shape[3]]
+            
         return x
 
+    def _pad_nonmultiple_input(self, x):
```

Comment:
This method is exactly the same in 2D and 2.5D models (`diff -u`):
```diff
@@ -227,6 +271,7 @@
         :return torch.tensor x_padded: zero-padded x
         :return tuple pad_shape: shape x was padded by
         """
+        
         # zero-pad non multiple of 2**num_blocks shapes
         assert self.training == False, (
                 "Padding and cropping to shape requires evaluation mode.. "
```
We should consider a base or mixin class for these.

Reply:
Yep. Been considering this from the start. Have avoided so far since there would only be two instances of the base-class type.

Reply:
We can also make this a utility function. There is only one integer parameter (`num_blocks`) needed from `self`, and it does not modify the state of the object in any way. I can do this if you don't have conflicting local changes.

Reply:
Actually I might take it further away into the inference module so that the switching logic is no longer in the model itself. See #214 for reasoning. @Christianfoley this will break your new tests and I'm open to any push back.

Reply:
Changed in https://github.com/mehta-lab/microDL/pull/210/commits/7157d1f622309d5b83f2e2b9f63975a0d5ddef3c